clang/test/Sema/attr-counted-by-late-parsed.c

```diff
@@ -0,0 +1,54 @@
+// RUN: %clang_cc1 -fsyntax-only -verify %s
+
+#define __counted_by(f)  __attribute__((counted_by(f)))
+
+struct size_unknown;
+
+struct at_pointer {
+  int count;
+  struct size_unknown *__counted_by(count) buf; // expected-error{{'counted_by' cannot be applied to an sized type}}
```

Comment:
TODO: emit an error for 'counted_by' on unknown size like this and update the error message here.

Reply:
This should be fine, because `count` is declared before use&mdash;unless I'm getting type attributes confused with field attributes.. Also, there's a grammar-o in the message: `to _an_ sized type`.